go/saltpack/decrypt.go

```diff
@@ -257,13 +258,19 @@ func (ds *decryptStream) processEncryptionBlock(bl *EncryptionBlock) ([]byte, er
 	}
 
 	nonce := ds.nonce.ForPayloadBox(blockNum)
+	ciphertext := bl.PayloadCiphertext
+	hash := sha512.Sum512(ciphertext)
 
-	tag, err := ds.tagKey.Unbox(nonce, bl.TagCiphertexts[ds.position])
+	hashBox, err := ds.tagKey.Box(nonce, hash[:])
 	if err != nil {
+		return nil, err
```

Comment:
Actually I'm not sure why `Box` returns an error value at all. How can we fail to box? Bad key lengths or something?